### ISSUE TITLE
[Insight] Object parameters should be type hinted

### ISSUE THREAD

**Issue by LukasHeinz** [shopsys/shopsys#333]
**[in packages/framework/src/Model/Order/Mail/OrderMailService.php, line 259](https://github.com/shopsys/shopsys/blob/f0212bec7fa7d1ae1ce1b5e5feb78b3b4593dc11/packages/framework/src/Model/Order/Mail/OrderMailService.php#L259)**

> The parameter `order`, which is an object, should be typehinted.

```php

    /**
     * @param \Shopsys\FrameworkBundle\Model\Order\Order $order
     * @return string
     */
    private function getDomainLocaleByOrder($order)
    {
        return $this->domain->getDomainConfigById($order->getDomainId())->getLocale();
    }
}

```

**[in packages/framework/src/Component/Setting/SettingValue.php, line 121](https://github.com/shopsys/shopsys/blob/f0212bec7fa7d1ae1ce1b5e5feb78b3b4593dc11/packages/framework/src/Component/Setting/SettingValue.php#L121)**

> The parameter `value`, which is an object, should be typehinted.

```php
    }

    /**
     * @param \DateTime|string|int|float|bool|null $value
     */
    protected function setValue($value)
    {
        $this->type = $this->getValueType($value);
        if ($this->type === self::TYPE_BOOLEAN) {
            $this->value = $value === true ? self::BOOLEAN_TRUE : self::BOOLEAN_FALSE;
        } elseif ($this->type === self::TYPE_NULL) {
```

**[in packages/framework/src/Model/Cookies/CookiesFacade.php, line 83](https://github.com/shopsys/shopsys/blob/f0212bec7fa7d1ae1ce1b5e5feb78b3b4593dc11/packages/framework/src/Model/Cookies/CookiesFacade.php#L83)**

> The parameter `cookiesArticle`, which is an object, should be typehinted.

```php

    /**
     * @param \Shopsys\FrameworkBundle\Model\Article\Article|null $cookiesArticle
     * @param int $domainId
     */
    public function setCookiesArticleOnDomain($cookiesArticle, $domainId)
    {
        $cookiesArticleId = null;
        if ($cookiesArticle !== null) {
            $cookiesArticleId = $cookiesArticle->getId();
        }
```

---------------------

_Posted from [SensioLabsInsight](https://insight.sensiolabs.com/projects/3b647b6b-ff37-406c-ae0a-e004328ef47f/analyses/1?status=violations)_

**Comment by LukasHeinz** [comment]
This is output from SensioLabs Insight and it is only suggestion. Please consider it.

**Comment by TomasLudvik** [comment]
Hello @LukasHeinz,

@MattCzerner has solved this issue in PR #351. The SettingValue::setValue has not been changed, because there are multiple data types accepted by this method, so it could not be type hinted.

**Comment by MattCzerner** [comment]
Hello fixed in #351
